DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs

```diff
@@ -423,342 +441,77 @@ private void RegisterCommands(IEnumerable<ApplicationCommandsModuleConfiguration
-                                DiscordApplicationCommandLocalization SubDescriptionLocalizations = null;
-
-                                if (translations != null)
-                                {
-                                    var command_translation = translations.Single(c => c.Name == payload.Name);
-                                    if (command_translation != null && command_translation.SubGroups != null)
-                                    {
-
-                                        var sub_command_translation = command_translation.SubGroups.Single(sc => sc.Name == subGroupAttribute.Name);
-
-                                        if (sub_command_translation != null)
-                                        {
-                                            SubNameLocalizations = sub_command_translation.NameTranslations;
-                                            SubDescriptionLocalizations = sub_command_translation.DescriptionTranslations;
-                                        }
-                                    }
-                                }
-
-                                //Similar to the one for regular groups
-                                foreach (var subsubmethod in subsubmethods)
-                                {
-                                    var suboptions = new List<DiscordApplicationCommandOption>();
-                                    var commatt = subsubmethod.GetCustomAttribute<SlashCommandAttribute>();
-                                    var parameters = subsubmethod.GetParameters();
-                                    if (parameters.Length == 0 || parameters == null || !ReferenceEquals(parameters.First().ParameterType, typeof(InteractionContext)))
-                                        throw new ArgumentException($"The first argument must be an InteractionContext!");
+                        if (!string.IsNullOrEmpty(ctx.Translations))
+                        {
+                            groupTranslations = JsonConvert.DeserializeObject<List<GroupTranslator>>(ctx.Translations);
+                        }
 
-                                    parameters = parameters.Skip(1).ToArray();
-                                    suboptions = suboptions.Concat(await this.ParseParameters(parameters, guildid)).ToList();
+                        var slashGroupsTulpe = NestedCommandWorker.ParseSlashGroupsAsync(type, classes, guildid, groupTranslations).Result;
 
-                                    DiscordApplicationCommandLocalization SubSubNameLocalizations = null;
-                                    DiscordApplicationCommandLocalization SubSubDescriptionLocalizations = null;
-                                    List<DiscordApplicationCommandOption> LocalizisedOptions = null;
+                        if (slashGroupsTulpe.Item1 != null && slashGroupsTulpe.Item1.Any())
+                            updateList.AddRange(slashGroupsTulpe.Item1);
```

Comment:
Honestly, in this context Item1, Item2 and so on is rather confusing. I recommend an alternative approach similar to tuple syntax you're using here

```csharp
(string name, string department) SampleGet()
{
    return("super awesome name", "super awesome department");
}

void SampleUse()
{
     var result = SampleGet();
     Console.WriteLine($"{result.name}: {result.department}");

   var (name, department) = SampleGet();
   Console.WriteLine($"{name}: {department}");
}
```